package/gluon-mesh-wireless-sae/luasrc/lib/gluon/upgrade/205-wireless-mesh-sae

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/lua
+
+local util = require 'gluon.util'
+local site = require 'gluon.site'
+local hash = require 'hash'
+local uci = require('simple-uci').cursor()
+
+
+local function configure_sae(vif, enable)
+	if enable then
+		uci:set('wireless', vif, 'encryption', 'sae')
+		uci:set('wireless', vif, 'key', site.wifi.mesh.sae_passphrase() or hash.md5(site.prefix6()))
+		uci:set('wireless', vif, 'ieee80211w', 2)
```

Comment:
The activation of MFP has to be done conditionally only when the device supports MFP.

This is a bit of a problem for devices, which do not support MFP, however this is most likely a small edge case, as only Atheros <9160 do not seem to support it.

Reply:
That basically affects the wr2543n/nd v1. Anything else?

Reply:
I guess I can make that contingent on `platform.supports_mfp` from #1876.

Reply:
The big question is now: What should be the default (off, optional, required)?

Depends on:
- which radios cannot handle mfp?
  - enabling mfp would exclude these devices/targets from wireless mesh
- is there a difference between optional and required for radios that cannot handle mfp?
  - as in: can we default to optional, because that gives us the best security
  - or: will setting mfp to optional already incur breakage

Reply:
In the kernel headers I even see a feature flag NL80211_EXT_FEATURE_MFP_OPTIONAL: "Driver supports the %NL80211_MFP_OPTIONAL value in %NL80211_ATTR_USE_MFP."

This might mean that there are drivers that will work with "required", but break with "optional"? Not entirely sure how to interpret this.

Reply:
I did not come across such a device (which doesn't mean there are such cases out there).

Honestly, i would not make much about it. MFP is required for WiFi 5 certification and (to my knowledge) all those drivers correctly support it. We are only cutting off AR9132 devices and ramips-rt305x as well as ramips-mt7620.

Looking at the ramips case, these devices cannot even so SAE. The AR9132 can so SAE without MFP, however this chip is flawed in many ways and the obscure TP-Link TL-WR2543 is the only device with 64M RAM we are supporting.

As SAE mesh only makes sense when using with 802.11w, i would propose to check 802.11w capability per-radio and enable SAE only in this case. When 802.11w is not supported, it should be disabled.

Thoughts on that?

Reply:
> i would propose to check 802.11w capability per-radio and enable SAE only in this case. 

Those devices would then be able to mesh among each other, but no with the remaining nodes of the network. I would rather want to signal, that if your node cannot do SAE it does not do wireless mesh at all.

Reply:
> When 802.11w is not supported, it should be disabled.

Is that ("it") wireless mesh in general or SAE in particular?